Suggest users prefer CYLC_SHARE_CYCLE_DIR to ROSE_DATAC (#2782)
metomi/rose: sphinx/tutorial/rose/furthertopics/date-time-manipulation.rst

   cylc cyclepoint --offset-hours=-3

The `isodatetime`_ command provides functionality

> beyond ``cylc cyclepoint``. Cylc also provides the ``CYLC_SHARE_CYCLE_DIR``

Reviewer: ```suggestion
beyond ``cylc cyclepoint``. Cylc also provides the ``CYLC_TASK_SHARE_CYCLE_DIR``
```

The ``ROSE_DATAC`` Environment Variable
---------------------------------------

.. note::

>    From Cylc 8.3.1 the variable ``CYLC_SHARE_CYCLE_DIR`` should be preferred

Reviewer: ```suggestion
   From Cylc 8.5.0 the variable ``CYLC_TASK_SHARE_CYCLE_DIR`` should be preferred
```

      <run directory>/share/cycle/<cycle>

   These are called the ``share/cycle`` directories.
   ``CYLC_SHARE_CYCLE_DIR`` provides the path to::

      "$CYLC_WORKFLOW_SHARE_DIR/cycle/$CYLC_TASK_CYCLE_POINT"

To get the path to a previous (or a future) ``share/cycle`` directory we can
provide an offset to isodatetime:

.. code-block:: bash

   isodatetime "${CYLC_TASK_CYCLE_POINT} --offset -P1D

Which can be used to provide custom offset directory locations:

.. code-block:: bash

   # Cylc task script
   CYCLE_POINT_MINUS_P1D=$(isodatetime "${CYLC_TASK_CYCLE_POINT} --offset -P1D)"
   SHARE_CYCLE_DIR_MINUS_P1D="${CYLC_WORKFLOW_SHARE_DIR}/cycle/${CYCLE_POINT_MINUS_P1D}"
   mkdir -p "${SHARE_CYCLE_DIR_MINUS_P1D}"

>    echo "I believe in ..." > "${SHARE_CYCLE_DIR_MINUS_P1D}/somefile"

Reviewer 2: I don't quite get this example.

Maybe show how to retrieve data from an earlier task rather than creating it.

Author: Se if you prefer the new version.

      <run directory>/share/cycle/<cycle>

   These are called the ``share/cycle`` directories.

>    ``CYLC_SHARE_CYCLE_DIR`` provides the path to::

Reviewer: ```suggestion
   ``CYLC_TASK_SHARE_CYCLE_DIR`` provides the path to::
```

      <run directory>/share/cycle/<cycle>

   These are called the ``share/cycle`` directories.
   ``CYLC_SHARE_CYCLE_DIR`` provides the path to::

      "$CYLC_WORKFLOW_SHARE_DIR/cycle/$CYLC_TASK_CYCLE_POINT"

To get the path to a previous (or a future) ``share/cycle`` directory we can
provide an offset to isodatetime:

.. code-block:: bash

   isodatetime "${CYLC_TASK_CYCLE_POINT} --offset -P1D

Which can be used to provide custom offset directory locations:

.. code-block:: bash

   # Cylc task script
   CYCLE_POINT_MINUS_P1D=$(isodatetime "${CYLC_TASK_CYCLE_POINT} --offset -P1D)"

>    TODAYS_DATA="${CYLC_WORKFLOW_SHARE_DIR}/cycle/${CYLC)TASK_CYCLE_POINT}/output.data"

Reviewer 2: Typo

```suggestion
   TODAYS_DATA="${CYLC_WORKFLOW_SHARE_DIR}/cycle/${CYLC_TASK_CYCLE_POINT}/output.data"
```

   YESTERDAYS_DATA="${CYLC_WORKFLOW_SHARE_DIR}/cycle/${CYCLE_POINT_MINUS_P1D}/output.data"

   # Here I'm imagining a day-to-day comparison program
   # (This is an example, not a built in Cylc program):

>    python -m compare.py "${YESTERDAYS_DATA}" "${TODAYS_DATA}"

Reviewer 2: We probs don't need to go this far.

Author: This?

```suggestion
   # Do something with "${YESTERDAYS_DATA}" and "${TODAYS_DATA}"
```

Reviewer 2: Pretty much.
